lists_hw/MyLinkedList.java

```diff
@@ -0,0 +1,110 @@
+package ListLab;
+
+import java.util.Iterator;
+
+public class MyLinkedList<T> implements Iterable<T>{
+
+	private Node<T> head;
+	private int size = 0;
+
+	public int getSize() {
+		return size;
+	}
+
+	public boolean remove(int index) {
+		//to-do
+		if (index > size) {
+			return false;
+		}
+		Node<T>node = head;
+		Node<T>afterNode;
+		for (int i = 0; i < index-1; i++) {
+			node = node.getNext();
+		}
+		afterNode = node.getNext();
+		node.setNext(afterNode);
+		return true;
+
+	}
+
+	public T get(int index) {
+		//to-do
+		Node<T> nodeAtIndex = head;
+		for (int i = 0; i <index ; i++) {
+			nodeAtIndex = nodeAtIndex.getNext();
+		}
+		return nodeAtIndex.getData();
+	}
+
+	public void add(T obj) {
+		//to-do
+		if (head == null) {
+			head = new Node<>(obj);
+		}
+		Node<T> node = head;
+		while (node.getNext()!=null) {
+			node = node.getNext();
+		}
+		return node.getData();
+	}
+
+	public void reverse () {
+		Node<T>node = head;
+		Node<T>node2 = head;
+
+        while(node.getNext() != null) {
+			node = node.getNext();
+		}
+
+		head = node;
+        
+        while (node != head){
+            while (node2.getNext().getNext() != null) {
+                node2 = node2.getNext();
+            }
+
+            node.setNext(node2);
+            node = node.getNext();
+            node2 = head;
+        }
```

Comment:
I think I see where you're going with this - move `node` to the end of the list, then set its next to be the second to last element, represented by `node2`, etc. You haven't worked out all the details, but you're on the right track.